url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/DeviceGraphicsDisplay/GantryGraphicsDisplay.java

```diff
@@ -0,0 +1,132 @@
+package DeviceGraphicsDisplay;
+
+import java.awt.Graphics2D;
+import java.util.ArrayList;
+
+import javax.swing.JComponent;
+
+import Networking.Client;
+import Networking.Request;
+import Utils.BinData;
+import Utils.Constants;
+import Utils.Location;
+
+public class GantryGraphicsDisplay extends DeviceGraphicsDisplay {
+
+	Location currentLocation;
+	Location destinationLocation;
+	Location binLocation;
+
+	ArrayList<BinGraphicsDisplay> binList;
+
+	boolean isBinHeld = false; // Determines whether or not the gantry is holding a bin
+	boolean isMoving = false;
+
+	private final int velocity = 5;
+
+	BinGraphicsDisplay heldBin;
+
+	BinData tempBin;
+
+	Client client;
+
+	public GantryGraphicsDisplay(Client c) {
+		currentLocation = new Location(Constants.GANTRY_ROBOT_LOC);
+		destinationLocation = new Location(Constants.GANTRY_ROBOT_LOC);
+		binList = new ArrayList<BinGraphicsDisplay>();
+		client = c;
+		tempBin = null;
+	}
+
+	@Override
+	public void draw(JComponent c, Graphics2D g) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 44-71 into a separate method.